frontend/webapp/containers/main/destinations/destination-form-body/test-connection/index.tsx

```diff
@@ -1,53 +1,67 @@
+type Status = 'success' | 'error';
+
+interface Props {
   destination: DestinationInput;
   disabled: boolean;
-  clearStatus: () => void;
+  status?: Status;
   onError: () => void;
+  onSuccess: () => void;
+  validateForm: () => boolean;
 }
 
-const ActionButton = styled(Button)<{ $success?: boolean }>`
+const ActionButton = styled(Button)<{ $status?: Status }>`
   display: flex;
   align-items: center;
   gap: 8px;
-  background-color: ${({ $success }) => ($success ? 'rgba(129, 175, 101, 0.16)' : 'transparent')};
-`;
 
-const ActionButtonText = styled(Text)<{ $success?: boolean }>`
-  font-family: ${({ theme }) => theme.font_family.secondary};
-  font-weight: 500;
-  text-decoration: underline;
-  text-transform: uppercase;
-  font-size: 14px;
-  line-height: 157.143%;
-  color: ${({ theme, $success }) => ($success ? theme.text.success : theme.colors.white)};
+  ${({ $status, theme }) =>
+    $status === 'success'
+      ? css`
+          border-color: transparent;
+          background-color: ${theme.colors.success};
+        `
+      : $status === 'error'
+      ? css`
+          border-color: transparent;
+          background-color: ${theme.colors.error};
+        `
+      : css`
+          background-color: transparent;
+        `}
 `;
 
-export const TestConnection: React.FC<TestConnectionProps> = ({ destination, disabled, clearStatus, onError }) => {
+export const TestConnection: React.FC<Props> = ({ destination, disabled, status, onError, onSuccess, validateForm }) => {
   const { testConnection, loading, data } = useTestConnection();
-  const success = useMemo(() => data?.testConnectionForDestination.succeeded || false, [data]);
 
   useEffect(() => {
     if (data) {
-      clearStatus();
-      if (!success) onError && onError();
+      const { succeeded } = data.testConnectionForDestination;
+
+      if (succeeded) onSuccess();
+      else onError();
     }
-  }, [data, success]);
+  }, [data]);
+
+  const onClick = async () => {
+    if (validateForm()) await testConnection(destination);
+  };
 
   return (
-    <ActionButton variant='secondary' disabled={disabled} onClick={() => testConnection(destination)} $success={success}>
-      {loading ? <FadeLoader /> : success ? <Image alt='checkmark' src={getStatusIcon('success')} width={16} height={16} /> : null}
+    <ActionButton $status={status} variant='secondary' disabled={disabled} onClick={onClick}>
+      {loading ? <FadeLoader /> : !!status ? <Image src={getStatusIcon(status)} alt='status' width={16} height={16} /> : null}
 
-      <ActionButtonText size={14} $success={success}>
-        {loading ? 'Checking' : success ? 'Connection OK' : 'Test Connection'}
-      </ActionButtonText>
+      <Text family='secondary' decoration='underline' size={14} color={!!status ? theme.text[status] : undefined}>
+        {loading ? 'Checking' : status === 'success' ? 'Connection OK' : status === 'error' ? 'Connection Failed' : 'Test Connection'}
+      </Text>
     </ActionButton>
   );
```

Comment:
I think we return the error message, maybe we can integrate it with the new error mechanism that you build